### ISSUE TITLE
instance template module: attempt to use `subnetwork_project` variabe fails with API error for not supported cross-project reference

### ISSUE THREAD

**Issue by ivankorn** [terraform-google-modules/terraform-google-vm#29]
There is some issue with implementation of `subnetwork_project` as cross-project reference fails with following error:

```
~/workspace/google/terraform-google-lb-http/examples/shared-vpc]$ terraform apply -auto-approve
module.cloud-nat.random_string.name_suffix: Refreshing state... [id=t24x9w]
data.template_file.group-startup-script: Refreshing state...
module.gce-lb-http.google_compute_firewall.default-hc[0]: Refreshing state... [id=group-http-lb-hc-0]
module.gce-lb-http.google_compute_http_health_check.default[0]: Refreshing state... [id=group-http-lb-backend-0]
module.gce-lb-http.google_compute_global_address.default: Refreshing state... [id=group-http-lb-address]
google_compute_network.default: Refreshing state... [id=shared-vpc-lb-http]
module.mig.data.google_compute_zones.available: Refreshing state...
module.mig_template.data.google_compute_image.image_family: Refreshing state...
module.mig_template.data.google_compute_image.image: Refreshing state...
google_compute_router.default: Refreshing state... [id=us-central1/lb-http-router]
google_compute_subnetwork.default: Refreshing state... [id=us-central1/shared-vpc-lb-http]
module.cloud-nat.google_compute_router_nat.main: Refreshing state... [id=us-central1/lb-http-router/cloud-nat-lb-http-router]
module.mig_template.google_compute_instance_template.tpl: Creating...

Error: Error creating instance template: googleapi: Error 400: Invalid value for field 'resource.properties.networkInterfaces[0].subnetwork': 'projects/gl-ik-shared/regions/us-central1/subnetworks/shared-vpc-lb-http'. Cross-project references for this resource are not allowed., invalid

  on .terraform/modules/mig_template/terraform-google-modules-terraform-google-vm-cb3d6cc/modules/instance_template/main.tf line 51, in resource "google_compute_instance_template" "tpl":
  51: resource "google_compute_instance_template" "tpl" {
```

**Comment by ocervell** [comment]
Any updates on this ? Do we know if the issue is in the API, Google provider, or Terraform module ?

**Comment by morgante** [comment]
Haven't tested it recently, but #52 might have fixed it. Can you dive in and take a look?

**Comment by aaron-lane** [comment]
@ocervell can you verify if this was fixed by #52?

**Comment by DDOlafGrote** [comment]
@aaron-lane It seems that the bug still exists. Do you have an idea or a current state of this bug?

We received the following error after `terraform apply`:

`google_compute_instance_template.appserver: Creating...
Error: Error creating instance template: googleapi: Error 400: Invalid value for field 'resource.properties.networkInterfaces[0].subnetwork': 'projects/shared-vpc/regions/europe-west4/subnetworks/subnet-name-xxx'. Cross-project references for this resource are not allowed., invalid`

We use the following provider versions:

```
provider "google" {
  region  = var.gcp_region
  project = var.gcp_project_id
  version = "= 3.15.0"
}
```

```
provider "google-beta" {
  region  = var.gcp_region
  project = var.gcp_project_id
  version = "= 3.15.0"
}
```

```
terraform {
  required_version = "= 0.12.23"
  ....
}
```

**Comment by aaron-lane** [comment]
@DDOlafGrote unfortunately I no longer maintain this module.

**Comment by DDOlafGrote** [comment]
@aaron-lane Thx for your feedback. For your information: I solved the problem by increasing the Google provider to 3.24.0. 

From my perspective, this issue will be fixed by Google provider 3.24.0. 

```
provider "google" {
  region  = var.gcp_region
  project = var.gcp_project_id
  version = "= 3.24.0"
}
 
provider "google-beta" {
  region  = var.gcp_region
  project = var.gcp_project_id
  version = "= 3.24.0"
}
```
